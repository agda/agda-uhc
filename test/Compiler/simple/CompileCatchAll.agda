{-# OPTIONS -v treeless:20 #-}
module _ where

open import Common.Prelude

f : List Nat → List Nat → Nat
f _        []       = 0
f []       (y ∷ ys) = y
f (x ∷ xs) _        = x

main : IO Unit
main = printNat (f [] [])
    ,, printNat (f (1 ∷ []) [])
    ,, printNat (f [] (2 ∷ []))
    ,, printNat (f (3 ∷ []) (4 ∷ []))
