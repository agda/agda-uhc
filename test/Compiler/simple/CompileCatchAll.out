EXECUTED_PROGRAM

ret > ExitSuccess
out > { compiling CompileCatchAll.f
out > CompileCatchAll.f =
out >   λ a b →
out >     case b of
out >       Agda.Builtin.List.List.[] → 0
out >       Agda.Builtin.List.List._∷_ c d →
out >         case a of
out >           Agda.Builtin.List.List.[] → c
out >           Agda.Builtin.List.List._∷_ e f → e
out > }
out > { compiling CompileCatchAll.main
out > CompileCatchAll.main =
out >   Common.IO.then
out >     () () _ _
out >     (Common.IO.printNat
out >        (CompileCatchAll.f
out >           Agda.Builtin.List.List.[] Agda.Builtin.List.List.[]))
out >     (Common.IO.then
out >        () () _ _
out >        (Common.IO.printNat
out >           (CompileCatchAll.f
out >              (Agda.Builtin.List.List._∷_ 1 Agda.Builtin.List.List.[])
out >              Agda.Builtin.List.List.[]))
out >        (Common.IO.then
out >           () () _ _
out >           (Common.IO.printNat
out >              (CompileCatchAll.f
out >                 Agda.Builtin.List.List.[]
out >                 (Agda.Builtin.List.List._∷_ 2 Agda.Builtin.List.List.[])))
out >           (Common.IO.printNat
out >              (CompileCatchAll.f
out >                 (Agda.Builtin.List.List._∷_ 3 Agda.Builtin.List.List.[])
out >                 (Agda.Builtin.List.List._∷_ 4 Agda.Builtin.List.List.[])))))
out > }
out > 0023
